### PR TITLE
Clean up WIN32/WindowsStore in CMake projects.

### PR DESCRIPTION
**Pull request by brycehutchings**
This simplifies things in most places by removing WindowsStore-specific
code or fixing areas that assume "Windows" system when they should
assume WIN32 (which includes WindowsStore). As a consequence, more
projects can be built for UWP. Finally, this fixes build issues when
building ARM/ARM64 for Win32 related to bad OpenGL and Vulkan
detection in CMake.

### CODE REVIEW

File: src/tests/list/CMakeLists.txt [hunk 1]
```diff
@@ -41,8 +41,6 @@ if(MSVC)
 endif()
 
 set_target_properties(runtime_list PROPERTIES FOLDER ${TESTS_FOLDER})
-set_target_properties(runtime_list PROPERTIES OUTPUT_NAME openxr_runtime_list)
```
Comment (rpavlik) [inline src/tests/list/CMakeLists.txt:44]: Why was this removed?

**Review by rpavlik — COMMENTED** [review]
Looks pretty good, just have one question where a change I made got reverted. I had namespaced the name of the runtime list app so it could be installed system wide on Linux. If that breaks some other platform I can carry the patch in the Debian packaging, but I'd also like to know what gets broken 😁

File: src/tests/list/CMakeLists.txt (hunk 1 above)
Comment (brycehutchings) [reply src/tests/list/CMakeLists.txt:44]: Yeah it broke the UWP packaging support in CMake because it puts the entrypoint executable name in as runtime_list and then it can't find the file (I guess it isn't polished enough to use OUTPUT_NAME if specified). Can we rename the entire target to openxr_runtime_list instead?

Comment (rpavlik) [reply src/tests/list/CMakeLists.txt:44]: Ah, weird, go figure. Yep, we can rename the whole thing, that works for me!

Comment (brycehutchings) [reply src/tests/list/CMakeLists.txt:44]: Okay renamed the whole target

**Comment by rpavlik** [comment]
awesome, thanks!

**Review by rpavlik — APPROVED** [review]
